core/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -464,164 +458,157 @@ public String toString() {
-        exceptions.put(org.elasticsearch.index.translog.TranslogException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.cluster.metadata.ProcessClusterEventTimeoutException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.action.support.replication.TransportReplicationAction.RetryOnPrimaryException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.ElasticsearchTimeoutException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.search.query.QueryPhaseExecutionException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.repositories.RepositoryVerificationException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.search.aggregations.InvalidAggregationPathException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.script.groovy.GroovyScriptExecutionException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.indices.IndexAlreadyExistsException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.script.Script.ScriptParseException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.transport.netty.SizeHeaderFrameDecoder.HttpOnTransportException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.index.mapper.MapperParsingException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.search.SearchContextException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.search.builder.SearchSourceBuilderException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.index.engine.EngineClosedException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.action.NoShardAvailableActionException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.action.UnavailableShardsException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.index.engine.FlushFailedEngineException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.common.breaker.CircuitBreakingException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.transport.NodeNotConnectedException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.index.mapper.StrictDynamicMappingException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.action.support.replication.TransportReplicationAction.RetryOnReplicaException.class, exceptions.size());
-        exceptions.put(org.elasticsearch.indices.TypeMissingException.class, exceptions.size());
+        final Map<Class<? extends ElasticsearchException>, IdAndCtor> exceptions = new HashMap<>();
+        exceptions.put(org.elasticsearch.index.snapshots.IndexShardSnapshotFailedException.class, new IdAndCtor(org.elasticsearch.index.snapshots.IndexShardSnapshotFailedException::new, exceptions.size()));
```

Comment:
Can you rewrite this as an array of ctor references and iterate the list to build the map? It might be a bit easier to read.